src/Makefile

```diff
@@ -29,6 +29,16 @@ endif
 WARN=-Wall -W -Wno-missing-field-initializers
 OPT=$(OPTIMIZATION)
 
+# Detect if the compiler supports C11 _Atomic
+C11_ATOMIC := $(shell sh -c 'echo "\#include <stdatomic.h>" > foo.c; \
```

Comment:
@ShooterIT  / @oranagra - This test does not actually work (at least not on Debian)  -- it always causes CC errors on missing the foo.c  file (which are subsequently ignored) and the code gets compiled with std=c99.
You can see this failure in the CI run: https://github.com/redis/redis/runs/1117499989

Reply:
@ushachar thanks!
we already noticed, but still trying to figure it out.
it seems to work on MacOS, but fail on Ubuntu.
from what i've seen so far if we remove the `rm foo.c` at the end, it passes.
and if i put `echo "pwd: `pwd`" >&2` instead, i notice that it runs twice.
maybe there's some kind of a race where one delete's the files before the other one uses it.
maybe you have good enough make skills to figure it out?

Reply:
Reason for the failure is that in Debian, 'sh' is Dash -- not Bash/Zsh.
Simply replacing 'sh' with /bin/bash fixes the issue on Debian, and should work on OSX/RHEL as well.

Reply:
found it.. it wasn't a race condition in the makefile, but a race in the short bash script: https://github.com/redis/redis/pull/7822. (Dash was faster than GCC)

Reply:
That's not a race - It's a syntax difference (around &>)

In bash, that redirects both stdout and err
In dash, it probably spawns a subshell
(fix in #7822 is good...)